include/seqan3/alignment/pairwise/execution/execution_handler_parallel.hpp

```diff
@@ -127,6 +128,33 @@ class execution_handler_parallel
         assert(status == contrib::queue_op_status::success);
     }
 
+    /*!\brief Takes underlying range of sequence pairs and invokes an alignment on each instance.
+     * \tparam algorithm_t              The type of the alignment algorithm.
+     * \tparam indexed_sequence_pairs_t The type of underlying sequence pairs annotated with an index;
+     *                                  must model seqan3::detail::indexed_sequence_pair_range.
+     * \tparam delegate_type            The type of the callable invoked on the std::invoke_result of `algorithm_t`.
+     *
+     * \param[in] algorithm              The alignment algorithm to invoke.
+     * \param[in] indexed_sequence_pairs The range of underlying annotated sequence pairs to be aligned.
```

Comment:
Ok, so it expects always a range. 
It never uses the range as a lvalue reference because it executes the alignments asynchronously and want's to make sure that another thread is not modifying this reference. And in the implementation we actually pass in a rvalue view so that nothing is expensive in this context.

Reply:
Ok. If you always pass a view in our context we might want to constrain this?
But no string feelings

Reply:
Well, the interface does not care really. It just takes a copy of whatever comes in. Just we use it by passing in a view which is lightweight. The only constraint I might add here is that the type is move constructible since afterwards we move it around.